### ISSUE TITLE
Differences in Tags are not calculated correctly

### ISSUE THREAD

**Issue by fbreckle** [terraform-providers/terraform-provider-aws#6363]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Hi,

terraform seems to recreate all Tags on an instance, even Tags that were unchanged.
This is a problem if recreation of a tag is not allowed by IAM permissions and in general unnecessary behavior. The aws API seems to silently succeed when creating duplicate tags.

### Terraform Version
$ terraform -v
Terraform v0.11.10
+ provider.aws v1.42.0

### Affected Resource(s)

* aws_instance (but anything that is getting tagged probably)

### Terraform Configuration Files

Start with

```hcl
provider "aws" {
  region = "eu-west-1"
  version = "1.42"
}

resource "aws_instance" "web" {
  ami           = "ami-0a5e707736615003c"
  instance_type = "t2.micro"

  tags {
    Origin = "Terraform"
  }
}
```

then add another tag "Testtag" to the instance as per 

```hcl
tags {
    Origin = "Terraform"
    Testtag = "Testy"
  }
```
### Debug Output
`
2018-11-05T18:21:17.617+0100 [DEBUG] plugin.terraform-provider-aws_v1.42.0_x4: 2018/11/05 18:21:17 [DEBUG] Creating tags: [{
2018-11-05T18:21:17.617+0100 [DEBUG] plugin.terraform-provider-aws_v1.42.0_x4:   Key: "Origin",
2018-11-05T18:21:17.617+0100 [DEBUG] plugin.terraform-provider-aws_v1.42.0_x4:   Value: "Terraform"
2018-11-05T18:21:17.617+0100 [DEBUG] plugin.terraform-provider-aws_v1.42.0_x4: } {
2018-11-05T18:21:17.617+0100 [DEBUG] plugin.terraform-provider-aws_v1.42.0_x4:   Key: "Testtag",
2018-11-05T18:21:17.617+0100 [DEBUG] plugin.terraform-provider-aws_v1.42.0_x4:   Value: "Testy"
2018-11-05T18:21:17.617+0100 [DEBUG] plugin.terraform-provider-aws_v1.42.0_x4: }] for <instance_id>
`

### Expected Behavior
Only the Tag "Testtag" should be created.

### Actual Behavior
Terraform creates both Tags anew, aborting if (re-)creation of the old Tag is forbidden.

### Steps to Reproduce
Modify the Tag list of an instance after creating it, then `terraform apply`

### Important Factoids
I only tested and verified this with EC2 Instances, but the source code makes me believe that many more, if not all, taggable elements on aws have this issue. I cant think of an element where this behavior is desirable/required.

I also have the faulty method fixed already, I will make a pull request tomorrow. I just want to make sure this is indeed a bug and Im not overlooking something.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 1.43.1 of the AWS provider, likely later today. 👍

**Comment by bflad** [comment]
This has been released in [version 1.43.1 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1431-november-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
